fv3core/stencils/updatedzd.py

```diff
@@ -233,6 +232,172 @@ def edge_profile_stencil(
+            largest_possible_shape, grid.compute_origin(add=(-self.grid.halo, 0, 0))
+        )
+        self._xfx_adv = utils.make_storage_from_shape(
+            largest_possible_shape, grid.compute_origin(add=(0, -self.grid.halo, 0))
+        )
+        self._yfx_adv = utils.make_storage_from_shape(
+            largest_possible_shape, grid.compute_origin(add=(-self.grid.halo, 0, 0))
+        )
+        self._ra_x = utils.make_storage_from_shape(
+            largest_possible_shape,
+            grid.compute_origin(add=(0, -self.grid.halo, 0)),
+        )
+        self._ra_y = utils.make_storage_from_shape(
+            largest_possible_shape,
+            grid.compute_origin(add=(-self.grid.halo, 0, 0)),
+            cache_key="updatedzd_ra_y",
+        )
+        self._wk = utils.make_storage_from_shape(
+            largest_possible_shape, grid.full_origin()
+        )
+        self._fx2 = utils.make_storage_from_shape(
+            largest_possible_shape, grid.full_origin()
+        )
+        self._fy2 = utils.make_storage_from_shape(
+            largest_possible_shape, grid.full_origin()
+        )
+        self._fx = utils.make_storage_from_shape(
+            largest_possible_shape, grid.full_origin()
+        )
+        self._fy = utils.make_storage_from_shape(
+            largest_possible_shape, grid.full_origin()
+        )
+        self._z2 = utils.make_storage_from_shape(
+            largest_possible_shape, grid.full_origin()
+        )
+
+        self.finite_volume_transport = FiniteVolumeTransport(
+            spec.namelist, spec.namelist.hord_tm
+        )
+        ax_offsets = fv3core.utils.axis_offsets(
+            self.grid, self.grid.full_origin(), self.grid.domain_shape_full()
+        )
+        self._ra_update = FixedOriginStencil(
+            ra_update,
+            origin=self.grid.full_origin(),
+            domain=self.grid.domain_shape_full(add=(0, 0, 1)),
+            externals=ax_offsets,
+        )
+        self._edge_profile = FixedOriginStencil(
+            edge_profile,
+            origin=self.grid.full_origin(),
+            domain=self.grid.domain_shape_full(add=(0, 0, 1)),
+            externals=ax_offsets,
+        )
+        self._zh_damp = FixedOriginStencil(
+            zh_damp,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.domain_shape_compute(add=(0, 0, 1)),
+        )
+
+    def __call__(
+        self,
+        dp0: FloatFieldK,
+        zs: FloatFieldIJ,
+        zh: FloatField,
+        crx: FloatField,
+        cry: FloatField,
+        xfx: FloatField,
+        yfx: FloatField,
+        wsd: FloatFieldIJ,
+        dt: float,
+    ):
+        """
+        Args:
+            dp0: ???
+            zs: ???
+            zh: ???
+            crx: Courant number in x-direction (??? what units)
+            cry: Courant number in y-direction (??? what units)
+            xfx: ???
+            yfx: ???
+            wsd: ???
+            dt: ???
```

Comment:
I would argue we know a lot of these, they have been documented in other code.  but can be picked up in a future PR

Reply:
I've documented most of them in the follow-on PR. Trying to document these was what led to doing that PR, because I didn't understand for sure what `zh`, `xfx`, and `yfx` are until yesterday. If it's OK, let's review those docstrings there.